### PR TITLE
Make RequestServer log less stuff

### PR DESCRIPTION
**Pull request by nico**
When loading github.com in browser and collecting a full system profile, and then looking at the profile in pv's flamegraph:

Before:
- 5070 samples in WebContent
- 623 samples in RequestServer (12.3% of WebContent)

After:
- 5413 samples in WebContent
- 504 samples in RequestServer (9.3% of WebContent)

The two profiles don't have exactly the same length so they can't be directly compared, but the ratio of RequestServer and WebContent should be comparable. This looks like a double-digit percentage speedup for RequestServer.

### CODE REVIEW

File: Userland/Libraries/LibHTTP/HttpJob.cpp [hunk 1]
```diff
@@ -24,13 +24,13 @@ void HttpJob::start(NonnullRefPtr<Core::Socket> socket)
     };
     m_socket->set_idle(false);
     if (m_socket->is_connected()) {
-        dbgln("Reusing previous connection for {}", url());
+        dbgln_if(CHTTPJOB_DEBUG, "Reusing previous connection for {}", url());
```
Comment (linusg) [inline Userland/Libraries/LibHTTP/HttpJob.cpp:27]: I wonder what a CHTTP is...

**Review by linusg — APPROVED** [review]
nice :)

File: Userland/Libraries/LibHTTP/HttpJob.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibHTTP/HttpJob.cpp:27]: Me too! Maybe a remnant from the olden times in serenity, when classes started with `C`?

Comment (alimpfard) [reply Userland/Libraries/LibHTTP/HttpJob.cpp:27]: Remnant of when it was `Core::HttpJob` (before just about the same time LibTLS came to be I think)

Comment (nico) [reply Userland/Libraries/LibHTTP/HttpJob.cpp:27]: dropping the `C` in #12084, thanks.

**Pull request merged**